### PR TITLE
Mtdsa 405 calculate total allowances and reliefs

### PR DESCRIPTION
**Pull request by dabd**

### CODE REVIEW

File: app/uk/gov/hmrc/selfassessmentapi/services/live/calculation/steps/SelfEmploymentProfitCalculation.scala [hunk 1]
```diff
@@ -21,7 +21,7 @@ import uk.gov.hmrc.selfassessmentapi.repositories.domain.{MongoLiability, MongoS
 
 object SelfEmploymentProfitCalculation extends CalculationStep {
 
-  private val annualInvestmentAllowance = 200000
+  private val annualInvestmentAllowance = BigDecimal(200000)
```
Comment (romankusnierz) [inline app/uk/gov/hmrc/selfassessmentapi/services/live/calculation/steps/SelfEmploymentProfitCalculation.scala]: This is acually integer, which gets implicitly converted to BigDecimal during calculations (if needed). I don't think we have to declare it as BigDecimal.

File: test/uk/gov/hmrc/selfassessmentapi/services/live/calculation/steps/TotalAllowancesAndReliefsSpec.scala [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    """calculate total allowances and reliefs by summing income tax relief
+      |across self-employment sources and adding the personal allowance""".stripMargin in {
+      val selfEmploymentId1 = "someSeId1"
+      val selfEmploymentId2 = "someSeId2"
+      val personalAllowance = BigDecimal(10000.00)
+      val lossBroughtForward1 = BigDecimal(550.00)
+      val lossBroughtForward2 = BigDecimal(450.00)
+      val liability = aLiability()
+        .copy(personalAllowance = Some(personalAllowance))
+        .copy(
```
Comment (romankusnierz) [inline test/uk/gov/hmrc/selfassessmentapi/services/live/calculation/steps/TotalAllowancesAndReliefsSpec.scala]: No need to call copy() twice

File: app/uk/gov/hmrc/selfassessmentapi/domain/Liability.scala [hunk 3]
```diff
@@ -99,6 +99,7 @@ object Liability {
         ),
         total = BigDecimal(3853.15)
       ),
-      totalTaxDue = BigDecimal(25796.95)
+      totalTaxDue = BigDecimal(25796.95),
+      totalAllowancesAndReliefs = BigDecimal(10000.00)
```
Comment (romankusnierz) [inline app/uk/gov/hmrc/selfassessmentapi/domain/Liability.scala]: Is this one of the fields we want to send back to API consumers? I cannot find it in the JIRA story, so I'm not sure.

File: app/uk/gov/hmrc/selfassessmentapi/services/live/calculation/steps/SelfEmploymentProfitCalculation.scala (hunk 1 above)
Comment (dabd) [inline app/uk/gov/hmrc/selfassessmentapi/services/live/calculation/steps/SelfEmploymentProfitCalculation.scala]: Fixed by https://github.com/hmrc/self-assessment-api/pull/118/commits/95eb6105d2818d003777a97e2e33d1c3b5c9ed88

File: test/uk/gov/hmrc/selfassessmentapi/services/live/calculation/steps/TotalAllowancesAndReliefsSpec.scala (hunk 2 above)
Comment (dabd) [inline test/uk/gov/hmrc/selfassessmentapi/services/live/calculation/steps/TotalAllowancesAndReliefsSpec.scala]: Fixed by https://github.com/hmrc/self-assessment-api/pull/118/commits/95eb6105d2818d003777a97e2e33d1c3b5c9ed88

**Pull request merged**